File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    words_d = dict()
    order = 0
    temp = words[0][0]
    for word in words:
        if word[0] != temp or word in words_d:
            break
        words_d[word] = 0
        order += 1
        temp = word[-1]
        
    answer = [0, 0] if len(words) == order else [order%n+1, int(order/n)+1]
    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.HashMap;

class Solution {
    public int[] solution(int n, String[] words) {
        HashMap<Integer, String> map = new HashMap<Integer, String>();
        int[] answer = new int[2];
        int i;
        
        map.put(0, words[0]);
        for (i = 1; i < words.length; i++) {
            String w = words[i];
            if (map.containsValue(w) || (w.charAt(0) != words[i-1].charAt(words[i-1].length()-1)))
                break;
            map.put(i, w);
        }
        
        if (map.size() == words.length) {
            answer[0] = 0;
            answer[1] = 0;
        } else {
            answer[0] = i%n+1;
            answer[1] = i/n+1;
        }
        
        return answer;
    }
}